### PR TITLE
[MU4] Fix build errors and compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: src/engraving/playback/renderers/arpeggiorenderer.cpp [hunk 1]
```diff
@@ -47,27 +47,28 @@ void ArpeggioRenderer::doRender(const Ms::EngravingItem* item, const mpe::Articu
         return;
     }
 
-    int stepsCount = chord->notes().size();
-    mpe::percentage_t percentageStep = mpe::HUNDRED_PERCENTS / stepsCount;
+    size_t stepsCount = chord->notes().size();
```
Comment (vpereverzev) [inline src/engraving/playback/renderers/arpeggiorenderer.cpp:50]: I suggest to just change this to `int stepsCount = static_cast<int>(chord->notes().size());` and not change anything else

Comment (Jojo-Schmitz) [reply src/engraving/playback/renderers/arpeggiorenderer.cpp:50]: OK

**Review by vpereverzev — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
hold a sec, I haven addition...

File: src/engraving/playback/renderers/arpeggiorenderer.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ void ArpeggioRenderer::doRender(const Ms::EngravingItem* item, const mpe::Articu
         return;
     }
 
-    int stepsCount = chord->notes().size();
+    int stepsCount = static_cast<int>(chord->notes().size());
```
Comment (Jojo-Schmitz) [inline src/engraving/playback/renderers/arpeggiorenderer.cpp:50]: Maybe better `static_cast<percentage_t>`?

File: src/framework/mpe/tests/multinotearticulationstest.cpp [hunk 3]
```diff
@@ -281,7 +281,8 @@ TEST_F(MultiNoteArticulationsTest, CrescendoPattern)
 
     for (size_t i = 0; i < noteCount; ++i) {
         // [GIVEN] Crescendo articulation applied on the note
-        ArticulationAppliedData crescendoApplied(crescendoMeta, 25 * ONE_PERCENT * i, 25 * ONE_PERCENT * (i + 1));
+        ArticulationAppliedData crescendoApplied(crescendoMeta, 25 * ONE_PERCENT * static_cast<int>(i),
+                                                 25 * ONE_PERCENT * (static_cast<int>(i) + 1));
```
Comment (Jojo-Schmitz) [inline src/framework/mpe/tests/multinotearticulationstest.cpp:285]: here too?

Comment (vpereverzev) [reply src/framework/mpe/tests/multinotearticulationstest.cpp:285]: you could just do the cast for the noteCount variable, I think it'll be enough here

Comment (Jojo-Schmitz) [reply src/framework/mpe/tests/multinotearticulationstest.cpp:285]: You mean `for (int i = 0; i < static_cast<int>(noteCount); ++i) {`?
That's not what we do further up, so not good for consistency

Comment (Jojo-Schmitz) [reply src/framework/mpe/tests/multinotearticulationstest.cpp]: Done. That got rid of a couple more casts

Comment (vpereverzev) [reply src/framework/mpe/tests/multinotearticulationstest.cpp]: `Or int noteCount = static_cast<int>(m_initialData.size());`

yes, this one

Comment (Jojo-Schmitz) [reply src/framework/mpe/tests/multinotearticulationstest.cpp]: Done. That got rid of a couple more casts

**Pull request merged**